jsjaws.py

```diff
@@ -709,13 +712,59 @@ def _flag_jsxray_iocs(output: Dict[str, Any], result: Result) -> None:
             jsxray_iocs_result_section.set_heuristic(2)
             result.add_section(jsxray_iocs_result_section)
 
-    def _extract_malware_jail_iocs(self, output: List[str], result: Result) -> None:
+    def parse_msdt_powershell(self, cmd):
+        import shlex
+
+        ori_parts = shlex.split(cmd)
+        parts = shlex.split(cmd.lower())
+
+        if "/param" in parts:
+            param = ori_parts[parts.index("/param") + 1]
+        elif "-param" in parts:
+            param = ori_parts[parts.index("-param") + 1]
+        else:
+            return cmd
+
+        for element in param.split():
+            if element.startswith("IT_BrowseForFile="):
+                return element[17:]
+        return cmd
+
+    def _extract_malware_jail_iocs(self, output: List[str], request: ServiceRequest) -> None:
         malware_jail_res_sec = ResultTableSection("MalwareJail extracted the following IOCs")
         for line in output:
             extract_iocs_from_text_blob(line, malware_jail_res_sec)
         if malware_jail_res_sec.body:
             malware_jail_res_sec.set_heuristic(2)
-            result.add_section(malware_jail_res_sec)
+            request.result.add_section(malware_jail_res_sec)
+
+        for line in self._parse_malwarejail_output(output):
+            log_line = line.split(" - ", 2)[1]
+            if log_line.startswith("location.href = "):
+                # We need to recover the non-truncated content from the sandbox_dump.json file
+                with open(self.malware_jail_sandbox_env_dump_path, "r") as f:
+                    data = load(f)
+                    location_href = data["location"]["_props"]["href"]
+                    if location_href.lower().startswith("ms-msdt:"):
+                        heur = Heuristic(5)
```

Comment:
rather than initializing the Heuristic class for the res, best practise would be to use the res.set_heuristic(5) method

Reply:
Why is this considered best practice if the option to pass a heuristic during initialization is there?
Is this considered best practice?
```python
heur = Heuristic(5)
res = ResultTextSection(heur.name)
res.set_heuristic(5)
request.result.add_section(res)
```